src/System.Memory/src/System/SpanHelpers.cs

```diff
@@ -76,6 +77,17 @@ private static bool IsReferenceOrContainsReferencesCore(Type type)
             return false;
         }
 
+        public static bool IsReferenceCore(Type type)
+        {
+            if (type.GetTypeInfo().IsPrimitive) // This is hopefully the common case. All types that return true for this are value types w/out embedded references.
```

Comment:
I doubt that this is saving anything. It can just return `type.GetTypeInfo().IsValueType`.